src/ucm/mmap/install.c

```diff
@@ -194,6 +221,36 @@ void ucm_fire_mmap_events(int events)
     ucm_fire_mmap_events_internal(events, &data);
 }
 
+static void ucm_mmap_event_report_missing(int expected, int actual)
+{
+    int events_count = 0;
+    int idx;
+    char *buf;
+    char *buf_p;
+    char *end_p;
+
+    buf = buf_p = malloc(UCM_MMAP_REPORT_BUF_LEN);
+    if (buf == NULL) {
+        ucm_error("Out of memory");
+        return;
+    }
+
+    end_p = buf_p + UCM_MMAP_REPORT_BUF_LEN;
+    ucs_for_each_bit(idx, expected) {
+        if (!(actual & UCS_BIT(idx))) {
+            snprintf(buf_p, end_p - buf_p, "%s%s", (events_count++ ? ", " : ""),
+                     ucm_mmap_event_name[idx]);
+            buf_p += strlen(buf_p);
+        }
+    }
+
+    if (events_count) {
+        ucm_diag("Missing events: %s", buf);
```

Comment:
ucm_diag("missing %s memory events: %s", title, buf);
where title is arg for ucm_mmap_event_report_missing, can be "internal" or "external"